### PR TITLE
add study def and plots to check new ICD10 covid codes

### PR DESCRIPTION
**Pull request by wjchulme**
PR to investigate impact of including new covid ICD10 codes.

The study definition retrieves covid-related unplanned hospital admissions and covid-related deaths after 3rd dose. There is a separate variable for each of the ICD-10 codes `[U071, U072, U099, U109]`, and for whether or not these are recorded in the "primary" slot for reason for admission or death. So that's `2*4*2=16` variables in total.

The analysis script defines further events where one of `[U071, U072]` were used (labelled `U07`) and where any of the above codes were used (labelled `all`). 

All these different outcome definitions are then cranked through `survfit` to get corresponding KM estimates. The estimates are rounded, and then plotted in a single graph to be inspected in the server to see whether we have an issue or not.

I suspect the plot might need some finessing once run on the real data.

### CODE REVIEW

**Review by elsie-h — APPROVED** [review]
All looks good. I've added two commits:

- [add ".R" to filename in project.yaml](https://github.com/opensafely/comparative-booster/pull/14/commits/524d9d71dce1a34d48459cc46d3132c42a647305), otherwise won't run via project.yaml
- [add function to summarise code combinations](https://github.com/opensafely/comparative-booster/pull/14/commits/381ca0a04fd38032cfdb2228982b68055181d739), as I'd be interested to know which codes are being used in combination (codes in "any" position, for admissions this is restricted to the first admission for each individual)

File: analysis/study_definition_ICD10check.py [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+  },
+  
+  #index_date = index_date,
+  
+  # This line defines the study population
+  population=patients.satisfying(
+    f"""
+      registered
+      AND
+      age >= 18
```
[the remark this answers is not in the document]
Comment (wjchulme) [reply analysis/study_definition_ICD10check.py:85]: I figured this doesn't matter very much at all

File: analysis/report_ICD10check.R [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+    sex = fct_case_when(
+      sex == "F" ~ "Female",
+      sex == "M" ~ "Male",
+      #sex == "I" ~ "Inter-sex",
+      #sex == "U" ~ "Unknown",
+      TRUE ~ NA_character_
+    ),
+
+    censor_date = pmin(dereg_date, anycovidvax_3_date+maxfup),
```
[the remark this answers is not in the document]
Comment (wjchulme) [reply analysis/report_ICD10check.R]: Think just to simplify things but it's easy to add back in, which I've done

**Comment by wjchulme** [comment]
Thanks, Elsie. I was wondering why, if the script action was missing the `.R` suffix, it wasn't picked up by github checks. But then I remembered I disabled them for this repo. Anyway, good spot.

Code for looking at code combinations looks good, thanks.

**Pull request merged**